server/game/cards/characters/02/croneofvaesdothrak.js

```diff
@@ -5,16 +5,16 @@ class CroneOfVaesDothrak extends DrawCard {
     setupCardAbilities(ability) {
         this.reaction({
             when: {
-                onCardDiscarded: (event, player, card, allowSave, originalLocation) =>
-                    ((originalLocation === 'hand' || originalLocation === 'draw deck')
-                     && card.getType() === 'character'
-                     && player !== this.controller)
+                onCardDiscarded: (event, params) =>
```

Comment:
I find this really hard to read.  params doesnt really mean anything, and is confused with the params a few lines down.

I don't know what the solution is, though.  Maybe One Day(tm) the event itself should just hold the relevant properties in a context-specific manner